gui/src/gshhs.cpp

```diff
@@ -1489,34 +1488,41 @@ int GshhsReader::selectBestQuality(ViewPort &vp) {
   return bestQuality;
 }
 
+// A singleton GshhsReader instance which is used to detect if a trajectory
+// crosses land.
+static GshhsReader *gshhs_singleton = NULL;
```

Comment:
Note: I've rename the static variable `reader` to `gshhs_singleton`. The `GSHHSChart` class also has a `reader` private member with the same name.

Reply:
Drive-by comment. That is not a singleton. You might want to  check the singleton  pattern  and use it or rename variable.

Reply:
Renamed variable